types.go

```diff
@@ -0,0 +1,39 @@
+package caddy
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var namespaceTypes map[string][]reflect.Type = make(map[string][]reflect.Type)
+
+func RegisterType(namespace string, vals []interface{}) {
```

Comment:
```suggestion
func RegisterNamespace(namespace string, vals []interface{}) {
```

How about this instead? A little more specific. With a comment that it maps namespaces to types expected from each namespace, solely for documentation purposes.

Reply:
Done.